doc/tutorials/basic/caching.md

```diff
@@ -0,0 +1,225 @@
+# Add Caching
+
+Caching allows us to store and reuse valuable computations, reducing the energy required for calculations and making our apps run faster and smoother:
+
+- Use `pn.cache` to *cache* function results.
+
+:::{note}
+When we ask to *run the code* in the sections below, we may execute the code directly in the Panel docs via the green *run* button, in a cell in a notebook, or in a file `app.py` that is served with `panel serve app.py --autoreload`.
+:::
+
+## Understand the Load
+
+To understand why caching can be helpful and how to apply it, its important to understand how a Panel application executes code.
+
+Create a file `external_module.py` containing the code below:
+
+```python
+print("running external_module.py")
+
+# I'm defined once and shared between all sessions (i.e. between all users)
+external_data={"__name__": __name__}
+```
+
+Create a file `app.py` containing the code below
+
+```python
+print("running app.py")
+import panel as pn
+
+from external_module import external_data
+
+pn.extension()
+
+# I'm defined each time the app loads and only shared within that session
+data = {"__name__": __name__}
+
+pn.Column(
+    "## External Module", external_data, f"Object id: {id(external_data)}",
+    "## App", data, f"Object id: {id(data)}",
+).servable()
+```
+
+Run
+
+```bash
+panel serve app.py
+```
+
+Open the app in your browser and refresh it a few times.
+
+It should look like
+
+<video controls="" poster="https://assets.holoviz.org/panel/tutorials/page_load_end.png">
+    <source src="https://assets.holoviz.org/panel/tutorials/page_load.mp4" type="video/mp4" style="max-height: 400px; max-width: 100%;">
+    Your browser does not support the video tag.
+</video>
+
+Please note how the `__name__` and `id` does not change for the `external_module` when you refresh the page. Note that they do change for the served `app`.
+
+:::{note}
+Please note
+
+- imported modules are run once the first time they are imported.
+  - Objects defined in imported modules are shared between all user *sessions*
+- the served `app.py` code is run every time the app is loaded.
+  - Objects defined here are shared within the single user *session* only (unless they are cached).
+- only *specific, bound functions* are rerun on user interactions. Not the entire `app.py` script.
+:::
+
+### Exercise: Add `--autoreload`
+
+Try repeating the steps above with `--autoreload`
+
+```bash
+panel serve app.py --autoreload
+```
+
+What changes?
+
+:::{dropdown} Solution
+
+With `--autoreload` both files are run when the server starts and before the page is loaded the first time.
```

Comment:
I did not either. But I realized there was a difference from creating these examples. I felt I had to explain this because otherwise users could get confused. Would be better (simpler) if with and without `--autoreload` executed the same.

Reply:
@philippjfr . Why does `--autoreload` have to run the code before any page is requested?